pkg/kubelet/apis/config/types.go

```diff
@@ -426,6 +426,15 @@ type KubeletConfiguration struct {
 	// +featureGate=MemoryQoS
 	// +optional
 	MemoryThrottlingFactor *float64
+	// registerWithTaints are an array of taints to add to a node object when
+	// the kubelet registers itself. This only takes effect when registerNode
+	// is true and upon the initial registration of the node.
+	// +optional
+	RegisterWithTaints []v1.Taint
```

Comment:
because this is now part of the config, and can have more values placed in it than the constrained CLI flag permitted, we need to validate this in `ValidateKubeletConfiguration`

* validate the individual fields similar to what is done in `parseTaint` (taint effect is valid, label value is valid, key is a qualified name, etc)
* ensure the TimeAdded field is nil

Reply:
@liggitt  PTAL
https://github.com/kubernetes/kubernetes/pull/105437/files#diff-b0fe5718080ac03df1496042447e4468534f1c4648522af21b064efdeee181b0R130